staging/src/k8s.io/kube-aggregator/pkg/registry/apiservice/strategy.go

```diff
@@ -92,10 +93,8 @@ func (apiServerStatusStrategy) PrepareForUpdate(ctx genericapirequest.Context, o
 	newAPIService := obj.(*apiregistration.APIService)
 	oldAPIService := old.(*apiregistration.APIService)
 	newAPIService.Spec = oldAPIService.Spec
-	newAPIService.Labels = oldAPIService.Labels
-	newAPIService.Annotations = oldAPIService.Annotations
-	newAPIService.Finalizers = oldAPIService.Finalizers
-	newAPIService.OwnerReferences = oldAPIService.OwnerReferences
+
+	metav1.ResetObjectMetaForStatus(newAPIService, oldAPIService)
```

Comment:
Shouldn't we actually test for modification in `ValidateUpdate` instread of overriding/resetting invalid field updates?